### ISSUE TITLE
temporarily failed to flush the buffer Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8

### ISSUE THREAD

**Issue by kenden** [fluent/fluentd-kubernetes-daemonset#13]
## Description

fluentd-kubernetes-daemonset does not send anything to the Cloudwatch group, it fails when trying to send with this error in the log:

```
2017-04-21 09:14:23 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:23 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `encode'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `to_json'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `block (2 levels) in write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:126:in `each'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:126:in `block in write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:104:in `each'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:104:in `write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:354:in `write_chunk'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:333:in `pop'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:342:in `try_flush'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:149:in `run'
```

This is the complete log

```
$ kubectl logs dev-fluentd-cloudwatch-035t9 -n logging
2017-04-21 09:13:16 +0000 [info]: reading config file path="/fluentd/etc/fluent.conf"
2017-04-21 09:13:16 +0000 [info]: starting fluentd-0.12.33
2017-04-21 09:13:17 +0000 [info]: gem 'fluent-mixin-config-placeholders' version '0.4.0'
2017-04-21 09:13:17 +0000 [info]: gem 'fluent-plugin-cloudwatch-logs' version '0.3.3'
2017-04-21 09:13:17 +0000 [info]: gem 'fluent-plugin-kubernetes_metadata_filter' version '0.26.3'
2017-04-21 09:13:17 +0000 [info]: gem 'fluent-plugin-record-reformer' version '0.9.0'
2017-04-21 09:13:17 +0000 [info]: gem 'fluent-plugin-secure-forward' version '0.4.3'
2017-04-21 09:13:17 +0000 [info]: gem 'fluentd' version '0.12.33'
2017-04-21 09:13:17 +0000 [info]: adding match pattern="fluent.**" type="null"
2017-04-21 09:13:17 +0000 [info]: adding filter pattern="kubernetes.**" type="kubernetes_metadata"
2017-04-21 09:13:21 +0000 [info]: adding match pattern="**" type="cloudwatch_logs"
2017-04-21 09:13:21 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: adding source type="tail"
2017-04-21 09:13:22 +0000 [info]: using configuration file: <ROOT>
  <match fluent.**>
    type null
  </match>
  <source>
    type tail
    path /var/log/containers/*.log
    pos_file /var/log/fluentd-containers.log.pos
    time_format %Y-%m-%dT%H:%M:%S.%NZ
    tag kubernetes.*
    format json
    read_from_head true
  </source>
  <source>
    type tail
    format /^(?<time>[^ ]* [^ ,]*)[^\[]*\[[^\]]*\]\[(?<severity>[^ \]]*) *\] (?<message>.*)$/
    time_format %Y-%m-%d %H:%M:%S
    path /var/log/salt/minion
    pos_file /var/log/fluentd-salt.pos
    tag salt
  </source>
  <source>
    type tail
    format syslog
    path /var/log/startupscript.log
    pos_file /var/log/fluentd-startupscript.log.pos
    tag startupscript
  </source>
  <source>
    type tail
    format /^time="(?<time>[^)]*)" level=(?<severity>[^ ]*) msg="(?<message>[^"]*)"( err="(?<error>[^"]*)")?( statusCode=($<status_code>\d+))?/
    path /var/log/docker.log
    pos_file /var/log/fluentd-docker.log.pos
    tag docker
  </source>
  <source>
    type tail
    format none
    path /var/log/etcd.log
    pos_file /var/log/fluentd-etcd.log.pos
    tag etcd
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/kubelet.log
    pos_file /var/log/fluentd-kubelet.log.pos
    tag kubelet
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/kube-proxy.log
    pos_file /var/log/fluentd-kube-proxy.log.pos
    tag kube-proxy
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/kube-apiserver.log
    pos_file /var/log/fluentd-kube-apiserver.log.pos
    tag kube-apiserver
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/kube-controller-manager.log
    pos_file /var/log/fluentd-kube-controller-manager.log.pos
    tag kube-controller-manager
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/kube-scheduler.log
    pos_file /var/log/fluentd-kube-scheduler.log.pos
    tag kube-scheduler
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/rescheduler.log
    pos_file /var/log/fluentd-rescheduler.log.pos
    tag rescheduler
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/glbc.log
    pos_file /var/log/fluentd-glbc.log.pos
    tag glbc
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <source>
    type tail
    format kubernetes
    multiline_flush_interval 5s
    path /var/log/cluster-autoscaler.log
    pos_file /var/log/fluentd-cluster-autoscaler.log.pos
    tag cluster-autoscaler
    format_firstline /^\w\d{4}/
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format %m%d %H:%M:%S.%N
  </source>
  <filter kubernetes.**>
    type kubernetes_metadata
  </filter>
  <match **>
    type cloudwatch_logs
    log_group_name k8s-testq
    auto_create_stream true
    use_tag_as_stream true
  </match>
</ROOT>
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-apiserver-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-8c84359f13adea39ed4a55a5125d1d0c0b7cd3a8d2cafa3bd28f693bafcd896a.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/etcd-server-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_etcd-container-53daba4bbd8bf05809af2b38de842804849a344f909da17a8b65592812c98f44.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/prometheus-prometheus-node-exporter-b3nkb_monitoring_prometheus-node-exporter-cd080c1c0153db79442f30f8203c650ece2cf461e903b63f75e5fb8549d6ba69.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-scheduler-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_kube-scheduler-fedcf148381b1376b40d02baab445a0cf6a9646fa864347b6d22662be6df585f.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/etcd-server-events-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_etcd-container-2b476b585a1fd76bd9aeb794ab768e7b517e8f19ef60683d4af55dcd17f9b678.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dns-controller-3905064033-7dpzl_kube-system_POD-52d24c9afa094d70aff656bd86ab80520f3c55d461598749d10606c4cbc58769.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-proxy-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-aab68b3db2ba54044fd14cca911c3a0c404816518d85897f8811b87dbe130a5b.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dev-fluentd-cloudwatch-035t9_logging_dev-fluentd-cloudwatch-295565aca55dfe7fe9dfacb6c55e693de6eedfd045a8c276593730ccf437e704.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/etcd-server-events-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-f4e74260b2e6060169bbd10a04cbae1d4a6537511033cd30ceb8c08a7b469517.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-controller-manager-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_kube-controller-manager-9b2ce2a7b180b03ddb6980dada16a75fd93e63791fddb710b902f51d8279310e.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dev-fluentd-cloudwatch-0cf0x_logging_dev-fluentd-cloudwatch-5f7aeeffb3e6837a17d9fc44cdcbbd71025dc9e09b1be54d4b8cbc72e487f643.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dns-controller-3905064033-7dpzl_kube-system_dns-controller-df93b047d0936b1cef30f3c4c7bb9004a0f51873279aeb25c7f31419c31f7f1f.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-controller-manager-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_kube-controller-manager-e33d89c54cec783c7b1c93a1e1c71851373a4331c25496351747a59369cbbe24.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-apiserver-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_kube-apiserver-c9d15e2dbbd44265d369cf26ecef754eb6fdf9ba26f7aa3648eacc8b0036735c.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dev-fluentd-cloudwatch-0cf0x_logging_POD-6788ad180875f93ced6ea831b2c32e3159219f6807de0b42184b66130554cd23.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/calico-node-8s1nt_kube-system_install-cni-4ef4fd7464bcd03261eb8a92089c773ba12bfed7fb4b8e616ad4aa58890f9046.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/etcd-server-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-fcb6afeb1431e326f918e8d007a3804f92030721fe0270487e86ea79899e2825.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-scheduler-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-912beb2045e51c78644cb32508f2da0f6186f82575b3f5457afbb8f7fcb0b046.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/prometheus-prometheus-node-exporter-b3nkb_monitoring_POD-ca353f808c94b2151b6ea49ddc77544612cf7eabbd185676eba3b41bf20f3261.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-proxy-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_kube-proxy-984aa615033c397dfe53a7d946f4aa340d3e9befc921be0165a6500ac611cf62.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/calico-node-8s1nt_kube-system_calico-node-1858f09a9872eb301f48df6a6854ed119adee7b26f042c3fd4f53f0c2a27da21.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/calico-node-8s1nt_kube-system_POD-595063d9b99efb2b6bb24289467e73350985f0e3c64e561a24cab3aac45fae51.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/dev-fluentd-cloudwatch-035t9_logging_POD-58a8d71f2511784ed483503cbaea39502de9257cefe9807d95f763ddb295750c.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/containers/kube-controller-manager-ip-10-0-104-242.eu-west-1.compute.internal_kube-system_POD-dbe4ddf33c9c294d1497bb5ce5859d66a9903f87d5ca110b2d34768e00862f73.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/etcd.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/kube-proxy.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/kube-apiserver.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/kube-controller-manager.log
2017-04-21 09:13:23 +0000 [info]: following tail of /var/log/kube-scheduler.log
2017-04-21 09:13:52 +0000 [info]: detected rotation of /var/log/containers/dev-fluentd-cloudwatch-0cf0x_logging_dev-fluentd-cloudwatch-5f7aeeffb3e6837a17d9fc44cdcbbd71025dc9e09b1be54d4b8cbc72e487f643.log; waiting 5 seconds
2017-04-21 09:13:52 +0000 [info]: detected rotation of /var/log/containers/dev-fluentd-cloudwatch-0cf0x_logging_POD-6788ad180875f93ced6ea831b2c32e3159219f6807de0b42184b66130554cd23.log; waiting 5 seconds
2017-04-21 09:14:23 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:23 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `encode'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `to_json'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:133:in `block (2 levels) in write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:126:in `each'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:126:in `block in write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:104:in `each'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.3.3/lib/fluent/plugin/out_cloudwatch_logs.rb:104:in `write'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:354:in `write_chunk'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:333:in `pop'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:342:in `try_flush'
  2017-04-21 09:14:23 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:149:in `run'
2017-04-21 09:14:24 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:26 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:24 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:14:26 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:30 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:26 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:14:30 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:37 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:30 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:14:38 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:14:53 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:38 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:14:54 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:15:21 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:14:54 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:15:22 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:16:19 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:15:22 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:16:19 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:18:37 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:16:19 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:18:38 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:23:09 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:18:38 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:23:09 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:32:17 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:23:09 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:32:17 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 09:49:24 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:32:17 +0000 [warn]: suppressed same stacktrace
2017-04-21 09:49:24 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 10:20:37 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 09:49:24 +0000 [warn]: suppressed same stacktrace
2017-04-21 10:20:37 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 11:29:40 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 10:20:37 +0000 [warn]: suppressed same stacktrace
2017-04-21 11:29:41 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-04-21 13:57:55 +0000 error_class="Encoding::UndefinedConversionError" error="\"\\xC2\" from ASCII-8BIT to UTF-8" plugin_id="object:2aeda49bb36c"
  2017-04-21 11:29:41 +0000 [warn]: suppressed same stacktrace
```

## Repro steps:

* Create a namespace so you can clean up easilly afterwards:  
  `kubectl create namespace encodingbug`

* Put the following into fluentd-cloudwatch.yaml
```yaml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  labels:
    app: fluentd-cloudwatch
    release: dev
  name: dev-fluentd-cloudwatch
spec:
  selector:
    matchLabels:
      app: fluentd-cloudwatch
      release: dev
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: fluentd-cloudwatch
        release: dev
    spec:
      containers:
      - env:
        - name: AWS_REGION
          value: eu-west-1
        - name: LOG_GROUP_NAME
          value: k8s-testq
        - name: AWS_ACCESS_KEY_ID
          valueFrom:
            secretKeyRef:
              key: aws_access_key_id
              name: dev-fluentd-cloudwatch
        - name: AWS_SECRET_ACCESS_KEY
          valueFrom:
            secretKeyRef:
              key: aws_secret_access_key
              name: dev-fluentd-cloudwatch
        image: fluent/fluentd-kubernetes-daemonset:v0.12.33-cloudwatch
        imagePullPolicy: IfNotPresent
        name: dev-fluentd-cloudwatch
        resources:
          limits:
            cpu: 100m
            memory: 200Mi
          requests:
            cpu: 100m
            memory: 200Mi
        terminationMessagePath: /dev/termination-log
        volumeMounts:
        - mountPath: /var/log
          name: varlog
        - mountPath: /var/lib/docker/containers
          name: varlibdockercontainers
          readOnly: true
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - hostPath:
          path: /var/log
        name: varlog
      - hostPath:
          path: /var/lib/docker/containers
        name: varlibdockercontainers
---
apiVersion: v1
data:
  aws_access_key_id: REDACTED_AWS_ACCESS_KEY_ID
  aws_secret_access_key: REDACTED_AWS_SECRET_ACCESS_KEY
kind: Secret
metadata:
  labels:
    app: fluentd-cloudwatch
    release: dev
  name: dev-fluentd-cloudwatch
type: Opaque
```

* Replace `REDACTED_AWS_ACCESS_KEY_ID` and `REDACTED_AWS_SECRET_ACCESS_KEY` by a valid AWS user with proper permissions in file `fluentd-cloudwatch.yaml`

* Run `kubectl create fluentd-cloudwatch.yaml --namespace encodingbug`

* Check the pods got created:  
  `kubectl get pods --namespace encodingbug -l "app=fluentd-cloudwatch,release=dev"`

* Check the logs:  
   This checks all pods:
  `kubectl  logs --namespace encodingbug -l "app=fluentd-cloudwatch,release=dev"`  
   You can (should) specify a specific pod:
  `kubectl  logs dev-fluentd-cloudwatch-035t9 --namespace encodingbug`

* Clean up step:  
  `kubectl delete namespace encodingbug`

## Suggestion:

Check this [Stackoverflow entry](http://stackoverflow.com/questions/18928721/encodingundefinedconversionerror-xc2-from-ascii-8bit-to-utf-8)

## Versions:

Docker image: `fluent/fluentd-kubernetes-daemonset:v0.12.33-cloudwatch`
Kubernetes: 1.5.2

**Comment by so0k** [comment]
Thanks for detailed report @edsiper - seems linked to https://github.com/ryotarai/fluent-plugin-cloudwatch-logs/issues/65

**Comment by alterx** [comment]
I'm having the same issue, looks like https://github.com/ryotarai/fluent-plugin-cloudwatch-logs/pull/58 could fix it, tho. Any ideas of why the PR haven't been merged?

**Comment by alterx** [comment]
Hey @kenden , I have a question:
I've been trying to deploy this daemon to integrate fluentd + CloudWatch with little success. 
Even though I've followed these instructions I keep getting this error:
https://github.com/fluent/fluentd-kubernetes-daemonset/issues/42

Have you ever seen it before? I'm running out of options here, maybe you can help

**Comment by alterx** [comment]
After a lot of tests it looks like the issue is related to the version of the `fluent-plugin-cloudwatch-logs` gem. If you update it to 0.4.0 the encoding error goes away.

I build a docker image using 

```
FROM fluent/fluentd:v0.12.33-debian

USER root
WORKDIR /home/fluent
ENV PATH /home/fluent/.gem/ruby/2.3.0/bin:$PATH

RUN buildDeps="sudo make gcc g++ libc-dev ruby-dev libffi-dev" \
     && apt-get update \
     && apt-get install \
     -y --no-install-recommends \
     $buildDeps \
    && echo 'gem: --no-document' >> /etc/gemrc \
    && gem install fluent-plugin-secure-forward \
    && gem install fluent-plugin-record-reformer \
    && gem install fluent-plugin-cloudwatch-logs -v 0.4.0 \
    && gem install fluent-plugin-kubernetes_metadata_filter \
    && gem install ffi \
    && gem install fluent-plugin-systemd -v 0.0.8 \
    && SUDO_FORCE_REMOVE=yes \
    apt-get purge -y --auto-remove \
                  -o APT::AutoRemove::RecommendsImportant=false \
                  $buildDeps \
 && rm -rf /var/lib/apt/lists/* \
    && gem sources --clear-all \
    && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem

# Copy configuration files
COPY ./conf/fluent.conf /fluentd/etc/
COPY ./conf/systemd.conf /fluentd/etc/
COPY ./conf/kubernetes.conf /fluentd/etc/

# Copy plugins
COPY plugins /fluentd/plugins/

# Environment variables
ENV FLUENTD_OPT=""
ENV FLUENTD_CONF="fluent.conf"

# jemalloc is memory optimization only available for td-agent
# td-agent is provided and QA'ed by treasuredata as rpm/deb/.. package
# -> td-agent (stable) vs fluentd (edge)
#ENV LD_PRELOAD="/usr/lib/libjemalloc.so.2"

# Run Fluentd
CMD exec fluentd -c /fluentd/etc/$FLUENTD_CONF -p /fluentd/plugins $FLUENTD_OPT
```

Uploaded it to the Docker Hub (https://hub.docker.com/r/alterx/fluentd-cloudwatch/) and it works like a charm. As you can see in the `Dockerfile` above, the only thing I changed was the the version of the `fluent-plugin-cloudwatch-logs` gem

**Comment by kenden** [comment]
@alterx Thanks for the feedback. there's a PR to update the version of the plugin:
https://github.com/fluent/fluentd-kubernetes-daemonset/pull/44

**Comment by alterx** [comment]
Perfect, will keep an eye on that PR :D

**Comment by kenden** [comment]
@alterx The PR was merged, the version of the ruby package [fluent-plugin-cloudwatch-logs](https://github.com/ryotarai/fluent-plugin-cloudwatch-logs) used is now 0.4.0.

**Comment by edsiper** [comment]
Fixed by #44
